client/my-sites/plans-grid/components/actions.tsx

```diff
@@ -397,16 +408,25 @@ const PlanFeaturesActionsButton: React.FC< PlanFeaturesActionsButtonProps > = (
 		'is-large-currency': isLargeCurrency,
 	} );
 
-	const handleUpgradeButtonClick = () => {
+	const handleUpgradeButtonClick = useCallback( () => {
 		if ( ! freePlan ) {
 			recordTracksEvent( 'calypso_plan_features_upgrade_click', {
 				current_plan: currentSitePlanSlug,
 				upgrading_to: planSlug,
 			} );
 		}
 
-		onUpgradeClick && onUpgradeClick();
-	};
+		onUpgradeClick?.();
+	}, [ currentSitePlanSlug, freePlan, onUpgradeClick, planSlug ] );
+
+	const handleFreeTrialButtonClick = useCallback( () => {
+		recordTracksEvent( 'calypso_plan_features_upgrade_click', {
+			current_plan: currentSitePlanSlug,
+			upgrading_to: freeTrialPlanSlug,
+		} );
+
+		onUpgradeClick?.( freeTrialPlanSlug );
+	}, [ currentSitePlanSlug, freeTrialPlanSlug, onUpgradeClick ] );
```

Comment:
Do we need a separate handler for this? I'd think `handleUpgradeButtonClick` should suffice for both cases.